core/src/main/java/org/apache/accumulo/core/iterators/OrIterator.java

```diff
@@ -39,29 +39,44 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * An iterator that handles "OR" query constructs on the server side. This code has been adapted/merged from Heap and Multi Iterators.
+ * An iterator that provides a sorted-iteration of column qualifiers for a set of column families in a row.
+ * It is important to note that this iterator <em>does not</em> adhere to the contract set forth by the
+ * {@link SortedKeyValueIterator}. It returns Keys in {@code row+colqual} order instead of
+ * {@code row+colfam+colqual} order. This is required for the implementation of this iterator (to work
+ * in conjunction with the {@code IntersectingIterator}) but is a code-smell. This iterator should only
```

Comment:
Very clear and concise but not pedantic.

> but is a code-smell

I don't think you have to insult the hygiene of our code unless you know of a use case where this design could cause problems.  Otherwise I would just say this iterator sorts differently by design.